### ISSUE TITLE
GraphQL authentifaction

### ISSUE THREAD

**Issue by joelisayas** [api-platform/api-platform#1129]
Hi,

Is there a way to implement authentication mutations with GraphQL ?
I'm trying to develop a react client app who needs  authentication (i'm using Apollo) and trying to develop something similar to this :
`const LOGIN_MUTATION = gql
   mutation LoginMutation($email: String!, $password: String!) {
    login(email: $email, password: $password) {
      token
    }
  }
`
Jwt is configured and works well with the REST Api. 

thanks for your advice

**Comment by teohhanhui** [comment]
I'm not at all familiar with GraphQL, but authentication is not mutation.

**Comment by joelisayas** [comment]
Yeah i thought too but i'm confuse with this sample code [https://www.howtographql.com/react-apollo/5-authentication/](url)

**Comment by teohhanhui** [comment]
It doesn't seem like they recommend doing anything different from a REST API for authentication: https://graphql.org/graphql-js/authentication-and-express-middleware/

**Comment by joelisayas** [comment]
So auth has to be made through REST and store the token on local storage ?

**Comment by soyuka** [comment]
Closing per #2987. Questions are best asked on [GitHub Discussions](https://github.com/api-platform/core/discussions) or [Stack Overflow (api-platform tag)](https://stackoverflow.com/questions/tagged/api-platform.com). The framework now lives at `api-platform/core`.
